File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //inputs for the text
    string text = get_string("Text: ");

    float letters = 0, words = 1, sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //counts the letters
        if (isalpha(text[i]))
        {
            letters++;
        }
        //counts the words
        if (isspace(text[i]))
        {
            words++;
        }
        //counts the sentences
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    //do the formula
    float l = letters / words * 100;
    float s = sentences / words * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;
    int indexx = round(index);
    
    //checks the grade
    if (indexx >= 1 && indexx <= 16)
    {
        printf("Grade %i\n", indexx);
    }
    else
    {
        if (indexx < 1)
        {
            printf("Before Grade 1\n");
        }
        if (indexx > 16)
        {
            printf("Grade 16+\n");
        }
    }
}

File: readability.py
import cs50


# inputs for the text
text = cs50.get_string("Text: ")

letters = 0
words = 0
sentences = 0
words = 1

for i in range(len(text)):
    # counts the letters
    if (text[i].isalpha()):
        letters += 1
    # counts the words
    elif (text[i].isspace()):
        words += 1
    # counts the sentences
    elif (text[i] == '.' or text[i] == '!' or text[i] == '?'):
        sentences += 1
# do the formula
l = float(letters / words * 100)
s = float(sentences / words * 100)
index = round((float)(0.0588 * l - 0.296 * s - 15.8))
    
# checks the grade
if (index >= 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    print("Grade ", index)
